Make gender customizable

luisa-beerboom opened the pull request:
Closes #3098

Elblinator reviewed (changes requested):
If a user had an assigned gender and this gender is deleted then the old gender is still displayed in the account/participant-detail view
and these users are not appearing with any gender filters

the following Error can appear if a gender which was there before is deleted: 
```
Error: Datastore service sends HTTP 400. The following locks were broken: 'user/can_change_own_password', 'user/committee_ids', 'user/committee_management_ids', 'user/default_number', 'user/default_password', 'user/default_structure_level', 'user/default_vote_weight', 'user/email', 'user/first_name', 'user/gender', 'user/id', 'user/is_active', 'user/is_physical_person', 'user/last_name', 'user/meeting_ids', 'user/meeting_user_ids', 'user/meta_deleted', 'user/option_ids', 'user/organization_id', 'user/password', 'user/title', 'user/username'
```

The visuals and the handling are fine by me, but @Eich-hoernchen, @MSoeb, @rrenkert  or @emanuelschuetze should confirm the visuals and handling after the change request is implemented and before merging

test instanz with demo data from 2023-12-13: client3101.staging.openslides.com

luisa-beerboom commented:
The filter thing _should_ be fixed now.

From your description I couldn't reproduce the other two bugs and I have no idea what may even cause them.
Please provide a proper reproduction description

luisa-beerboom commented:
Opened issue OpenSlides/openslides-backend#2070 for backend part

Elblinator reviewed (approved):
function

@emanuelschuetze or @MSoeb: Please confirm that these visuals/handling of the new feature are alright

MSoeb commented:
Found some issues in the PR:

- [ ] Account list: If a gender options have changed, added or deleted the original entries in the Account filter options are still available without any change.
- [ ] User create in Accounts or Participants: If a gender options have changed, added or deleted the original entries are still available without any change.
- [ ] Meeting > Participants > detail view and Filter option: Same problem like the two above.

General questions:
- How do we handle changed values for existing accounts/participants? - Possible solutions: If value has changed, the new value is added. If value is deleted, the gender switches to no selection.
- Is this okay? Did I miss something?

luisa-beerboom commented:
>     * [ ]  Account list: If gender options have changed, added or deleted the original entries in the Account filter options are still available without any change.
> 
>     * [ ]  User create in Accounts or Participants: If gender options have changed, added or deleted the original entries are still available without any change.
> 
>     * [ ]  Meeting > Participants > detail view and Filter option: Same problem like the two above.

Those should all work already and they actually do when I try them out. Are you sure you _saved_ the new gender settings in the settings menu?
Committing a new gender requires that the add button behind the input field is pressed and then the save button for the form itself is pressed. Deleting genders, too, requires the settings to be saved.

Please check again and if it still doesn't work, please give a description of how exactly you are interacting with the interface to produce the results you are getting.

> General questions:
> 
>     * How do we handle changed values for existing accounts/participants? - Possible solutions: If value has changed, the new value is added. If value is deleted, the gender switches to no selection.

What happens currently is that the deletion of a gender empties the gender field for all users that had it. Given that gender names can't change (they can only be deleted/added), there is no necessity to replace any values on users

Does that answer your questions?

emanuelschuetze commented:
Tested and discussed with abs team:
We don't like this implementation because custom gender can't edit. Additional it's a very low priority - and there is no customer request for this feature. So we decided to close this PR without merge. Maybe we can reopen it later.

jsangmeister commented:
> Tested and discussed with abs team: We don't like this implementation because custom gender can't edit.

Then we would need to implement a dedicated gender model. AFAIK, this was not wanted when we first discussed this issue.

> there is no customer request for this feature.

But a strong community request, see e.g. https://github.com/OpenSlides/openslides-client/issues/2207
